src/components/NavigationBar/type.tsx

```diff
@@ -0,0 +1,247 @@
+import React, { FunctionComponent, useState } from "react";
+import { Icon } from "react-feather";
+import { NavLink } from "react-router-dom";
+
+export interface NavigationBarProps {
+  navigationItems: NavigationItem[];
+}
+
+export enum NavigationItemType {
+  NavigationLink = "NavigationLink",
+  DropDownList = "DropDownList",
+  LabelButton = "LabelButton",
+  IconButton = "IconButton",
+}
+
+interface NavigationLink {
+  schema: NavigationItemType.NavigationLink;
+  id: string;
+  label: string;
+  className?: string;
+  path: string;
+}
+
+interface DropDownList extends Omit<NavigationLink, "schema"> {
+  schema: NavigationItemType.DropDownList;
+  dropdownItems: { id: string; label: string; path: string }[];
+}
+
+interface LabelButton extends Omit<NavigationLink, "schema"> {
+  schema: NavigationItemType.LabelButton;
+  onClick?: () => void;
+}
+
+interface IconButton extends Omit<NavigationLink, "schema"> {
+  schema: NavigationItemType.IconButton;
+  icon: Icon;
+}
+
+export type NavigationItem = NavigationLink | DropDownList | LabelButton | IconButton;
+
+export const NavigationBarItem: FunctionComponent<{ item: NavigationItem; onClick: (isOn: boolean) => void }> = ({
+  item,
+  onClick,
+}) => {
+  switch (item.schema) {
+    case NavigationItemType.IconButton:
+      return <IconButton item={item} onClick={onClick} />;
+    case NavigationItemType.LabelButton:
+      return <LabelButton item={item} onClick={onClick} />;
+    case NavigationItemType.DropDownList:
+      return <DropDownList item={item} onClick={onClick} />;
+    default:
+      return <NavigationLink item={item} onClick={onClick} />;
+  }
+};
+
+const NavigationLink: FunctionComponent<{ item: NavigationLink; onClick: (isOn: boolean) => void }> = ({
+  item,
+  onClick,
+}) => {
+  if (item.path.indexOf("http://") == 0 || item.path.indexOf("https://") == 0) {
+    return (
+      <a
+        className={`font-medium ${item.className}`}
+        href={item.path}
+        onClick={() => {
+          onClick(false);
+        }}
+      >
+        {item.label}
+      </a>
+    );
+  } else {
+    return (
+      <NavLink
+        className={`font-medium ${item.className}`}
+        to={item.path}
+        onClick={() => {
+          onClick(false);
+        }}
+      >
+        {item.label}
+      </NavLink>
+    );
+  }
+};
+
+const LabelButton: FunctionComponent<{ item: LabelButton; onClick: (isOn: boolean) => void }> = ({ item, onClick }) => {
+  if (item.path.indexOf("http://") == 0 || item.path.indexOf("https://") == 0) {
+    return (
+      <a href={item.path} className="w-full">
+        <button
+          className={`font-bold py-2 px-3 ${item.className}`}
+          data-testid={item.id}
+          onClick={() => {
+            onClick(false);
+            {
+              if (typeof item.onClick === "function") {
+                item.onClick();
+              }
```

Comment:
this part is weird. also, if it's not function probably will fail silently

Reply:
It is meant to fail silently if the item.onClick is null